File: Cdb/src/main/java/org/eso/ias/cdb/pojos/LogLevelDao.java
package org.eso.ias.cdb.pojos;

import ch.qos.logback.classic.Level;

/**
 * The hibernate mapping for the log level 
 * (as defined by the minimum set of log levels recognized by the system 
 * according to slf4j).
 * 
 * @author acaproni
 * 
 */
public enum LogLevelDao {
	OFF(Level.OFF),
    ERROR(Level.ERROR),
    WARN(Level.WARN),
    INFO(Level.INFO),
    DEBUG(Level.DEBUG),
    TRACE(Level.TRACE),
    ALL(Level.ALL);

    /**
     * The corresponding level in logback
     */
	private final Level logbackLogLevel;

    /**
     * Constructor
     *
     * @param logbackLogLevel The logback Level
     */
	private LogLevelDao(Level logbackLogLevel) {
	    this.logbackLogLevel=logbackLogLevel;
    }

    /**
     * Returns the lcorresponding level in logback that can be useful
     * for programmatically change the log level.
     *
     * @return The logback log level
     */
	public Level toLoggerLogLevel() {
	    return logbackLogLevel;
    }
}

File: Cdb/src/main/python/IasCdb/Dao/LogLevelDao.py
from enum import Enum, unique

@unique
class LogLevelDao(Enum):
    '''
    Valid log levels
    '''
    OFF=1
    ERROR=2
    WARN=3
    INFO=4
    DEBUG=5
    TRACE=6
    ALL=7

    @classmethod
    def from_string(cls, level: str):
        if not level:
            raise ValueError("Invalid empty log level")
        return getattr(LogLevelDao, level)
    
    def to_string(self) -> str:
        return self.name
